libvast/src/experimental_table_slice.cpp

```diff
@@ -685,7 +682,8 @@ experimental_table_slice::experimental_table_slice(
   state_.layout = type{chunk::copy(as_bytes(*slice_.layout()))};
   VAST_ASSERT(caf::holds_alternative<record_type>(state_.layout));
   auto decoder = record_batch_decoder{};
-  state_.record_batch = decoder.decode(slice.arrow_ipc());
+  state_.record_batch = decoder.decode(
+    as_arrow_buffer(parent->slice(as_bytes(*slice.arrow_ipc()))));
```

Comment:
It looks like there's an overload ` Status Consume(const uint8_t *data, int64_t size)` as well, so I'm wondering if it's necessary to bother with creating a buffer at all? Does the decoded record batch still know about the buffer it was created from?

Reply:
That overload calls `arrow::Buffer::Wrap` internally, and then causes the record batch to slice its internal buffer from the result of that, causing the lifetimes not to be shared correctly. We used this API before this PR.